modules/icon/icon.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+// This file was ported from https://github.com/Claudiohbsantos/github-material-icons-extension/blob/ff97e50980/src/lib/replace-icon.js
+// to go with small changes.
+
+package icon
+
+import (
+	"os"
+	"path"
+	"strings"
+
+	"code.gitea.io/gitea/modules/git"
+	"code.gitea.io/gitea/modules/json"
+)
+
+type Map struct {
+	FileNames      map[string]string `json:"fileNames"`
+	FolderNames    map[string]string `json:"folderNames"`
+	FileExtensions map[string]string `json:"fileExtensions"`
+	LanguageIds    map[string]string `json:"languageIds"`
+	Light          Light             `json:"light"`
+}
+
+type Light struct {
+	FileNames      map[string]string `json:"fileNames"`
+	FolderNames    map[string]string `json:"folderNames"`
+	FileExtensions map[string]string `json:"fileExtensions"`
+}
+
+var iconMap = Map{
+	FileNames:      make(map[string]string),
+	FolderNames:    make(map[string]string),
+	FileExtensions: make(map[string]string),
+	LanguageIds:    make(map[string]string),
+	Light:          Light{},
+}
+
+func Init() error {
+	data, err := os.ReadFile("assets/material-icons.json")
+	if err != nil {
+		return err
+	}
+
+	err = json.Unmarshal(data, &iconMap)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// EntryIcon returns the icon for the given git entry
+func EntryIcon(entry *git.TreeEntry) string {
```

Comment:
Maybe not everyone like the colorful icons, could we keep an app.ini configuration to switch?

Reply:
What about `[ui].FILE_ICONS = file-specific|basic` as key with `file-specific` as default value?

Reply:
If it's not too hard I will try to include it in the user's appearance settings. Do you (or anyone else) have a high-level instruction list on how to add a setting to the user's appearance settings page? (settings? forms? templates? migrations?)